src/main/java/org/cactoos/iterator/TailOf.java

```diff
@@ -49,11 +50,13 @@
     public TailOf(final int num, final Iterator<T> iterator) {
         this.origin = new LinkedList<>(
             new CollectionOf<>(
-                new HeadOf<>(
-                    num,
-                    new LinkedList<>(
-                        new CollectionOf<>(iterator)
-                    ).descendingIterator()
+                new IterableOf<>(
+                    new HeadOf<>(
```

Comment:
@mazdack I think you should be able to remove `new CollectionOf<>(new IterableOf<>(`: `HeadOf` is itself already a collection I suppose.

Reply:
nope, `HeadOf` is an iterator

Reply:
@mazdack ok